widgetry/src/backend_glow.rs

```diff
@@ -353,3 +353,202 @@ impl PrerenderInnards {
         self.window_adapter.draw_finished(gfc_ctx_innards)
     }
 }
+
+/// Uploads a sprite sheet of textures to the GPU so they can be used by Fill::Texture and
+/// friends to paint shapes.
+///
+/// `path` - image file which is a grid of images.
+/// `sprite_length` - the width and height of an individual cell in the image grid
+///
+/// The image file can have any number of sprites, but they must all be the same size.
+///
+/// Once uploaded, textures are addressed by their id, starting from 1, from left to right, top to
+/// bottom, like so:
+///
+///   ┌─┬─┬─┐
+///   │1│2│3│
+///   ├─┼─┼─┤
+///   │4│5│6│
+///   ├─┼─┼─┤
+///   │7│8│9│
+///   └─┴─┴─┘
+///
+/// Texture(0) is reserved for a pure white (no-op) texture.
+///
+/// Implementation is based on the the description of ArrayTextures from:
+/// https://www.khronos.org/opengl/wiki/Array_Texture.
+pub fn load_textures(
```

Comment:
Nice! Did you have to do anything besides copy this code from native and copy the shader code? Was there anything from native that didn't work on web?

Reply:
Nope, just the straightforward move. `image` compiles fine on wasm, and we're doing the IO with `slurp_file`, which uses the filesystem or `include_dir` underneath.